static/js/user_groups.ts

```diff
@@ -195,3 +201,26 @@ export function is_user_in_group(user_group_id: number, user_id: number): boolea
     }
     return false;
 }
+
+export function get_realm_user_groups_for_dropdown_list_widget(
+    exclude_internet_group: boolean,
+): UserGroupForDropdownListWidget[] {
+    return settings_config.system_user_groups_list
+        .filter((group) => {
+            if (!exclude_internet_group) {
+                return true;
+            }
+
+            return group.name !== "@role:internet";
+        })
+        .map((group) => {
+            const user_group = get_user_group_from_name(group.name);
+            if (!user_group) {
+                throw new Error(`Unknown group name: ${group.name}`);
+            }
+            return {
+                name: group.display_name,
+                value: user_group.id.toString(),
```

Comment:
We may need to change this function when we will add more settings and different settings will have different options. Need to think about how to do it in the best possible way such that the code is readability.